src/lib/components/ui/Modal/Modal.jsx

```diff
@@ -297,29 +285,11 @@ Modal.propTypes = {
    */
   position: PropTypes.oneOf(['top', 'center']),
   /**
-   * How to treat the modal when its content is too long to fit the screen. The `body` option
-   * turns on scrolling just for the modal content, while the `modal` option enables scrolling of
-   * the entire modal, including the header and the footer.
-   */
-  scrollMode: PropTypes.oneOf(['body', 'modal']),
-  /**
-   * Custom CSS for the end scrolling shadow.
-   */
-  scrollViewEndShadowStyle: PropTypes.shape({
-    background: PropTypes.string,
-    boxShadow: PropTypes.string,
-  }),
-  /**
-   * Size of the scrolling shadows.
-   */
-  scrollViewShadowSize: PropTypes.string,
-  /**
-   * Custom CSS for the start scrolling shadow.
+   * The `ScrollView` component to be used as a wrapper of the content to provide body scrolling functionality.
+   * If provided only the modal body will be scrollable.
+   * If set to `null` the entire modal including header and footer will be scrollable.
    */
-  scrollViewStartShadowStyle: PropTypes.shape({
-    background: PropTypes.string,
-    boxShadow: PropTypes.string,
-  }),
+  scrollView: PropTypes.node,
```

Comment:
We certainly want just a single `ScrollView`, if any. Could we narrow it down to a single element?

```suggestion
  scrollView: PropTypes.element,
```

Reply:
done